src/sqlite.rs

```diff
@@ -50,24 +61,25 @@ impl DB for SQLite {
         Ok(())
     }
 
-    fn read(&self, table: &str, key: &str, result: &mut HashMap<String, String>) -> Result<()> {
+    async fn read(&self, table: String, key: String) -> Result<HashMap<String, String>> {
         // TODO: cache prepared statement
         let mut sql = SqlBuilder::select_from(table);
         sql.field("*");
         // TODO: fields
         sql.and_where(format!("{} = :{}", PRIMARY_KEY, PRIMARY_KEY));
         let sql = sql.sql()?;
-        let mut stmt = self.conn.prepare(sql)?;
+        let conn = self.conn.lock().unwrap();
```

Comment:
It seems that SQLite doesn't have native async support. We should run all of the following operations in tokio's thread pool by `spawn_blocking`.